src/components/withIntl.js

```diff
@@ -0,0 +1,69 @@
+import React, {Component, createContext} from 'react';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import invariant from 'invariant';
+import {invariantIntlContext} from '../utils';
+
+function getDisplayName(Component) {
+  return Component.displayName || Component.name || 'Component';
+}
+
+const IntlContext = createContext(null);
+const {
+  Consumer: IntlConsumer,
+  Provider: IntlProvider
+} = IntlContext
+
+export const Provider = IntlProvider
+export const Consumer = IntlConsumer
```

Comment:
@DragonRaider5 do you think the `IntlContext` object should be exported directly instead of the Consumer? You won't be able to use the `useContext` hook if you only have access to the Consumer.

From https://reactjs.org/docs/hooks-reference.html#usecontext

> Correct: `useContext(MyContext)`
> Incorrect: `useContext(MyContext.Consumer)`
> Incorrect: `useContext(MyContext.Provider)`

Reply:
+1 to this — please do export the entire context object for use with React hooks!

Reply:
Nice catch, thanks for the feedback! I've not used hooks yet so I didn't now about this.

Changed it!